src/parser.rs

```diff
@@ -1319,6 +1319,8 @@ pub fn parse_function(
                                 "expected '('".to_string(),
                                 tokens[*index].span,
                             )));
+
+                            return (ParsedFunction::new(FunctionLinkage::Internal), error);
```

Comment:
We don't want to immediately return unless we're at the end of the input. Otherwise, we should tolerate errors and keep parsing.

Reply:
Uhm, okay. So the wanted behavior when `(` is missing is to go on and try parsing function parameters anyway?
What if the previous operation of parsing the generic parameters failed? If that fails, do we want to immediately return at line [1304](https://github.com/SerenityOS/jakt/pull/278/files/6cd18ef93cad3db5ca49339222519d4da6ea88e8#diff-4a04259da480a6b794a2e947e4cc03eff4d1aa9330836f5b91cac68c5398193fR1304)?
If not, I would imagine `index` to be left midway in the generic parameters section, how does it make sense to try and keep going? We should at least advance until we find the closing `>`, shouldn't we?
Also if `(` is missing, we should indeed fire an error, but if we don't want to return early then it would be good to advance tokens until we hopefully find `(`, and then go on from there, right?

Sorry, I'm not getting what the wanted behavior is at the moment.